### DISCUSSION TITLE
raw.githubusercontent.com doesnt load for me

### DISCUSSION THREAD

**Discussion by marshelx** [community/community#65687] [Other Feature Feedback, Questions, & Ideas]
Bug

Hello, I am in Russia, and for some reason when I try to access raw.githubusercontent.com, it simply doesn't respond and times out (PR_CONNECT_RESET_ERROR). I use Firefox, and I have tried accessing it on other browsers such as Chrome, Brave, Chromium, even Tor and it still returns the same error. Raw github is very important for me and I wanna know if there is a fix to this issue. Normal github loads fine.

Please help

**Comment by marshelx** [comment]
This issue appeared like 4 days ago, and since then I have tried to find a fix but didn't find one.

**Comment by orazmyradov** [comment]
It seems you're experiencing difficulty accessing `raw.githubusercontent.com` from Russia. There could be various reasons for this, but based on your description, it sounds like this domain might be blocked or restricted in Russia.

Here are some potential solutions to bypass such restrictions:

1. **VPN (Virtual Private Network):**
    - A VPN allows you to create a secure connection to another network over the Internet and can be used to bypass regional restrictions. There are many VPN providers available, and some of them offer free plans. Once connected to a VPN server located outside Russia, you should be able to access `raw.githubusercontent.com`.

2. **Web Proxies:**
    - Web proxies allow you to browse the internet through an intermediary server. This can also be used to bypass regional restrictions. There are many free web proxy services available.

3. **DNS Over HTTPS (DoH) or DNS Over TLS (DoT):**
    - This involves encrypting your DNS requests, which may bypass certain types of DNS-based blocking. Firefox has a built-in option for DNS Over HTTPS that you can enable:
        - Go to `Preferences` > `General` > scroll down to `Network Settings` > `Settings`.
        - Enable the option for `Enable DNS over HTTPS`.
        - Select a provider from the dropdown or use a custom one.

4. **Alternative Domain Services:**
    - There are services that mirror content from GitHub. One of the popular ones is `jsDelivr`. Instead of accessing content from `raw.githubusercontent.com`, you can use `cdn.jsdelivr.net`. 
    - For example, if you normally access a file using:
    ```
    https://raw.githubusercontent.com/user/repo/branch/file
    ```
    - With jsDelivr, it would be:
    ```
    https://cdn.jsdelivr.net/gh/user/repo@branch/file
    ```

5. **Host File Modifying:**
    - Sometimes, the block can be due to DNS-level redirection. You can try adding an entry for `raw.githubusercontent.com` to your hosts file. Find an IP address for `raw.githubusercontent.com` using a service like "WhatsMyDNS" and then add an entry to the hosts file (`/etc/hosts` on Linux or `C:\Windows\System32\drivers\etc\hosts` on Windows).

6. **Tor Browser:**
    - You mentioned you tried Tor, but ensure you have the latest version and have configured it correctly. With Tor, your connection gets relayed through several nodes, which makes it harder for regional restrictions to block your requests.

Remember, always use these tools responsibly and make sure you're aware of any legal implications or terms of service violations associated with bypassing regional restrictions or network blocks.

**Reply by marshelx** [reply]
Hello, can you please describe how to modify the hosts file and maybe that could be my solution. Also, sometimes raw github works for like 5 minutes and then stops working again.
